src/roswire/common/package.py

```diff
@@ -193,10 +204,17 @@ def from_paths(app_instance: 'AppInstance',
                     raise
             else:
                 packages.append(package)
-        return PackageDatabase(packages)
+        return PackageDatabase(packages, paths)
 
-    def __init__(self, packages: Collection[Package]) -> None:
+    def __init__(self,
+                 packages: Collection[Package],
```

Comment:
Small optimisation: `packages` could actually be an `Iterable[Package]`, that would let us pass a generator instead (avoiding the need to create redundant lists).